tls/doc/tls_introduction.md

```diff
@@ -0,0 +1,263 @@
+
+That being said, there are cases where client certificates are indeed useful. Mainly in machine-to-machine communication. IT departments often use their own CAs to issue client certificates that they hand out to developers of applications that access company services via TLS. The application then has to present its client certificate during the handshake phase. Only if the service can successfully authenticate the client via its client certificate, will it accept the TLS connection.
+
+### Truststores and Keystores
+
+The words "keystore" and "truststore" are often used as synonyms. This is not precise. Truststores are a specific subset of keystores containing someone else’s public keys or certificates. If you install root CA certificates on a machine, they go into a truststore.
+
+In the Java world the distinction is very strict. When the official documentation says "keystore," it really only refers to the place where you store your own keys.
+
+### Disabling authentication
+
+If you’ve read the article up to this point, you’ve probably already guessed that foregoing authentication is a bad idea. As mentioned, services tend to authenticate their consumers on the application level because this is more convenient than client certificates.
+
+It is a whole different story, though, if clients switch off certificate checks. What that means is that the client trusts whatever TLS certificate the service presents — and that obviously is a huge security hole. Unfortunately, it’s also a common practice. Users who don’t understand how certificates work or find importing certificates into their truststore too complicated are tempted to switch off certificate checking completely.
+
+This behavior is also fueled by the fact that a lot of internet forum entries suggest setting the corresponding configuration options as a quick fix without explaining the dangers. The users are happy because their TLS connection now works and often do not realize that it is not secure anymore.
+
+Why isn’t it secure? If you don’t check the validity of the certificate presented by a service, you create the perfect opportunity for a man-in-the-middle attack. This means the attacker does not even have to go through the trouble of trying to get a fake root CA certificate installed on your machine.
+
+## TLS Versions and Cipher Suites
+
+IT security evolves. Both the defensive part and the attacker side. Some mechanisms that once were an industry standard are now considered unsecure.
+
+How can this happen?
+
+Remember when we said that cryptography is based on the idea that reversing it is needs to be hard enough? We now have a lot more computing power in our hands, and new mathematical algorithms still are invented that reduce the effort of the calculations involved from unrealistic to manageable. Also, the designers of security mechanisms are human and as such sometimes plant fundamental flaws in a design that make an update unavoidable.
```

Comment:
`Remember when we said that cryptography is based on the idea that reversing it is needs to be hard enough?`
at first glance this sentence was hard to parse, i think that maybe the "it" & assumed noun antecedent conflict with the behavior we're describing.

`Remember when we said that purpose of encrypted data (or other appropriate participle noun combo) was to make it hard to do` or something like that. Sorry, don't have a clear suggestion. Think folks can also figure out what's meant too, so we can leave as is.